### ISSUE TITLE
chrome 70: every test fails on driver.maximizewindow()

### ISSUE THREAD

**Issue by filiprichtarik** [golemhq/golem#130]
with updating to chrome 70 we have issue with running tests
open browser action contains driver.maximizewindow() which worked until chrome 69 and still works on every other browser

but after the update to 70 we are getting this error

WebDriverException: Message: unknown error: failed to change window state to maximized, current state is normal (Session info: chrome=70.0.3538.77) (Driver info: chromedriver=2.43.600229 (3fae4d0cda5334b4f533bede5a4787f7b832d052),platform=Mac OS X 10.13.6 x86_64)

**Comment by danielmaddern** [comment]
I've seen something similar on Linux too, but its with resizing the window after its opened... This is most likely a bug in chrome or chromedriver, not golem. 

Have you tried ChromeDriver 2.42 to see if this helps on OSX?

**Comment by filiprichtarik** [comment]
yes, i agree that its an issue with chrome/chromedriver 

tried changing selenium versions (3.13 & 3.14.1), chromedrivers (2.41, 2.42 & 2.43) and chrome versions  (69 & 70), nothing has changed  - with 69 working, with 70 not

interesting observation .. with chrome 69, driver.maximize_window() was executed without glitch but did nothing (window did not resize). with chrome 70 the execution fails

plus there should be a change in golem logic too, condition to call driver.maximize_window() if driver is not headless, cuz it doesn't make sense

**Comment by danielmaddern** [comment]
https://bugs.chromium.org/p/chromedriver/issues/detail?id=2389&q=&colspec=ID%20Status%20Pri%20Owner%20Summary 

This is the defect for this issue.. I'm guessing this will be fixed in the next chromedriver release..

**Comment by filiprichtarik** [comment]
i'm guessing it wont be fixed (at least not soon), because in the last 3 comments, they are discussing that it's osx feature to not enable window size manipulation meaning they cannot fix it on their own (issue since 2014)

i propose fast fix in our code - wrapping driver.maximize_window() with condition " if driver is not headless OR os is not osx "

**Comment by filiprichtarik** [comment]
#133

**Comment by lucianopuccio** [comment]
I think it's fine to keep this condition until this bug is fixed, if ever.

I've added a setting to enable/disable maximizing the browser on start.

Thanks @filiprichtarik for the fix.

**Comment by r-roos** [comment]
Hi Gents, 

The work around currently implemented does affect the following situation:
- Test execution done from via macos in combination with a remote browser (Zalenium in my case, providing a chrome browser)

As the workaround checks for OS == Darwin it will not maximize the window. I am not sure if this is desired in case you have a remote browser

**Comment by filiprichtarik** [comment]
ou, my bad, didn't think about this combination, but i know how to fix it
i'll send one more PR for this, tomorrow

**Comment by r-roos** [comment]
Reopening this issue for two reasons:
1. There is still a PR open to improve the current workaround: #139 
2. Once chromedriver 2.44 is released it should be assessed if this workaround can be removed

**Comment by luciano-renzi** [comment]
I added a comment on #139 with what I think is the proper condition to check. If this works then the error won't happen but... Chrome will not open maximized in mac.

For **Chrome local** (i.e.: ```golem run project test -b chrome```) the setting 'start_maximized' should work in mac because it's using the ChromeOption argument 'start-maximized'. See browser.py line 82 and this: http://chromedriver.chromium.org/capabilities

For **Chrome remote** this works:
```
{
"remote_browsers": {
    "chrome_mac": {
        "browserName": "chrome",
        "platform" : "MAC",
        "goog:chromeOptions": {
            "args": ["start-maximized"]
         }
    }
}}
```

And this can be added to the node config as well.

**Comment by luciano-renzi** [comment]
fixed in 0.6.2
